python/pyarrow/_fs.pyx

```diff
@@ -1027,6 +1027,23 @@ cdef class PyFileSystem(FileSystem):
     ----------
     handler : FileSystemHandler
         The handler object implementing custom filesystem behavior.
+
+    Examples
+    --------
+    Create an fsspec-based filesystem object for Google Cloud Storage:
```

Comment:
Looks like this fails the CI documentation tests as `gcsfs` isn't installed.

Reply:
Awesome, Github filesystem will be a great example. Thanks!!
And sorry for asking for review before waiting for the CI to finish - was a bit too sure of myself =)